### PR TITLE
Separate the BCI version check

### PR DESCRIPTION
**Pull request by grisu48**
To increase visibility for outdated container test runs, split the container version check into a separate test run.

- Related ticket: https://progress.opensuse.org/issues/135365
- Verification runs: [podman](https://duck-norris.qe.suse.de/tests/14164#step/bci_version_check/32) | [docker](https://duck-norris.qe.suse.de/tests/14165#)

The db data push error is unrelated (see https://progress.opensuse.org/issues/137819)

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]

File: tests/containers/bci_version_check.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use utils qw(zypper_call script_retry);
+use db_utils qw(push_image_data_to_db);
+use containers::common;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal;
+
+    return unless (get_var('CONTAINER_IMAGE_TO_TEST') && get_var('CONTAINER_IMAGE_BUILD'));
```
Comment (mloviska) [inline tests/containers/bci_version_check.pm:24]: This should rather end up in the `lib/main_common.pm`

**Review by mloviska — APPROVED** [review]

File: tests/containers/bci_prepare.pm [hunk 2]
```diff
@@ -120,37 +120,6 @@ sub run {
...
-        }
-        if (get_var('IMAGE_STORE_DATA')) {
-            my $size_b = script_output("$engine inspect --format \"{{.VirtualSize}}\" $image");
-            my $size_mb = $size_b / 1000000;
-            record_info('Size', $size_mb);
-            push_image_data_to_db('containers', $image, $size_mb, flavor => get_required_var('BCI_IMAGE_MARKER'), type => 'VirtualSize');
-        }
-
-    }
 }
```
Comment (m-dati) [inline tests/containers/bci_prepare.pm:123]: Why, instead of a new module, not to move the code in a new bci_prepare function and call by `use ...`

File: tests/containers/bci_version_check.pm (hunk 1 above)
Comment (m-dati) [reply tests/containers/bci_version_check.pm:24]: To me seems that, if the `unless` check was intended to end the test, it should run before the termnal setting: isn't it?

File: tests/containers/bci_version_check.pm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+use Mojo::Base qw(consoletest);
+use utils qw(zypper_call script_retry);
+use db_utils qw(push_image_data_to_db);
+use containers::common;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal;
```
Comment (m-dati) [inline tests/containers/bci_version_check.pm:22]: This above also is in the `bci_prepare.pm`: could it be removed from here?

File: tests/containers/bci_version_check.pm (hunk 1 above)
Comment (m-dati) [reply tests/containers/bci_version_check.pm:24]: Moreover the new `unless` logic is different from the original 'if` check where the only CONTAINER_IMAGE_TO_TEST missing caused the full code block to skip and the CONTAINER_IMAGE_BUILD missing only skipped an inner sub-block.

Comment (grisu48) [reply tests/containers/bci_version_check.pm:24]: You both are right in your own way, but that was not the intention of this check.

Leaving this here allows to clone the test run and modify/delete the corresponding values without having the cloned test fail. This is especially useful when disabling `CONTAINER_IMAGE_BUILD`, and likely also for `CONTAINER_IMAGE_TO_TEST`.

I still like @mloviska suggestion and will add it to `lib/main_containers` additionally. But I prefer to have this check here as well, because in the past I encountered some corner cases, where the scheduling-based exclusion was not working as expected.

File: tests/containers/bci_version_check.pm (hunk 3 above)
Comment (grisu48) [reply tests/containers/bci_version_check.pm:22]: No, every test should ensure that the proper terminal is set.

Otherwise if you would change the module in `bci_prepare`, you would need to know which other test runs depend on it for doing so.

**Review by pdostal — APPROVED** [review]

**Comment by pdostal** [comment]
rocket
